storage/internal/benchmarks/client_pool.go

```diff
@@ -156,20 +174,41 @@ func getClient(ctx context.Context, api benchmarkAPI) *storage.Client {
 // mutex on starting a client so that we can set an env variable for GRPC clients
 var clientMu sync.Mutex
 
-func initializeHTTPClient(ctx context.Context, writeBufferSize, readBufferSize int, json bool) (*storage.Client, error) {
+// Client config
+type clientConfig struct {
+	writeBufferSize, readBufferSize int
+	useJSON                         bool // only applicable to HTTP Clients
+	setGCSFuseOpts                  bool // only applicable to HTTP Clients
+	connectionPoolSize              int  // only applicable to GRPC Clients
+}
+
+func initializeHTTPClient(ctx context.Context, config clientConfig) (*storage.Client, error) {
 	opts := []option.ClientOption{}
 
-	if writeBufferSize != useDefault || readBufferSize != useDefault {
+	if config.writeBufferSize != useDefault || config.readBufferSize != useDefault || config.setGCSFuseOpts {
 		// We need to modify the underlying HTTP client
-
 		base := http.DefaultTransport.(*http.Transport).Clone()
-		base.MaxIdleConnsPerHost = 100 // this is set in Storage as well
-		base.WriteBufferSize = writeBufferSize
-		base.ReadBufferSize = readBufferSize
 
-		http2Trans, err := http2.ConfigureTransports(base)
-		if err == nil {
-			http2Trans.ReadIdleTimeout = time.Second * 31
+		if config.setGCSFuseOpts {
+			base = &http.Transport{
+				MaxConnsPerHost:     100,
+				MaxIdleConnsPerHost: 100,
+				// This disables HTTP/2 in transport.
+				TLSNextProto: make(
+					map[string]func(string, *tls.Conn) http.RoundTripper,
+				),
+			}
+		}
+
+		base.MaxIdleConnsPerHost = 100 // this is set in Storage as well
+		base.WriteBufferSize = config.writeBufferSize
+		base.ReadBufferSize = config.readBufferSize
+
+		if !config.setGCSFuseOpts {
+			http2Trans, err := http2.ConfigureTransports(base)
```

Comment:
Where are we using http2Trans? This looks un-used. We can move this block to else part of line:192 if block.

Reply:
It's a bit of a confusing API but `http2.ConfigureTransports` returns a pointer to the transport config. Thus we can modify the values that will be used by the client without having to do pass http2trans back in anywhere.

Reply:
Ohh, got it.